Java/Ch 17. Hard/Q17_21_Volume_of_Histogram/QuestionD.java

```diff
@@ -0,0 +1,67 @@
+package Q17_21_Volume_of_Histogram;
+
+import java.util.Stack;
+
+public class QuestionD {
+
+	public static int computeHistogramVolume(int[] histogram) {
+		int waterVolume = 0;
+		if (histogram == null || histogram.length <= 1) {
+			return waterVolume;
+		}
+		Stack<Bar> bars = new Stack<>();
+		for (int barIndex = 0; barIndex < histogram.length; barIndex++) {
```

Comment:
The entire calculation will happen at max 2n. since elements in stack are getting removed.